lib/maxwell/middleware/logger.ex

```diff
@@ -16,47 +16,105 @@ defmodule Maxwell.Middleware.Logger do
   use Maxwell.Middleware
   require Logger
 
+  @levels [:info, :debug, :warn, :error]
+
   def init(opts) do
-    check_opts(opts)
-    Keyword.get(opts, :log_level, :info)
+    case Keyword.pop(opts, :log_level) do
+      {_, [_|_]} ->
+        raise ArgumentError, "Logger Middleware Options doesn't accept wrong_option (:log_level)"
+      {nil, _}                           -> [default: :info]
+      {options, _} when is_list(options) -> parse_opts(options)
+      {level, _}                         -> parse_opts([{level, :default}])
+    end
   end
 
-  def call(request_env, next_fn, level) do
+  def call(request_env, next_fn, options) do
```

Comment:
I think `call` function is concisely enough, if you still think there're too many logic into it, we can make another pr to fix it.
that's how plug do time format https://github.com/elixir-lang/plug/blob/master/lib/plug/logger.ex#L45-L46